tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -176,31 +179,13 @@ object NativeConfig {
     def withOptimize(value: Boolean): NativeConfig =
       copy(optimize = value)
 
-    override def withLinktimeProperties(v: Map[String, Any]): NativeConfig = {
-      def isNumberOrString(value: Any) = {
-        def hasSupportedType = value match {
-          case _: Boolean | _: Byte | _: Char | _: Short | _: Int | _: Long |
-              _: Float | _: Double | _: String =>
-            true
-          case _ => false
-        }
-
-        value != null && hasSupportedType
-      }
-
-      val invalid = v.collect {
-        case (key, value) if !isNumberOrString(value) => key
-      }
-      if (invalid.nonEmpty) {
-        System.err.println(
-          s"Invalid link-time properties: \n ${invalid.mkString(" - ", "\n", "")}"
-        )
-        throw new BuildException(
-          "Link-time properties needs to be non-null primitives or non-empty string"
-        )
-      }
+    def linktimeProperties: Map[String, Any] = {
+      predefinedLinktimeProperties(this) ++ customLinktimeProperties
+    }
 
-      copy(linktimeProperties = v)
+    override def withLinktimeProperties(v: Map[String, Any]): NativeConfig = {
+      checkLinktimeProperties(v)
+      copy(customLinktimeProperties = v)
     }
```

Comment:
I don't like that there is `linktimeProperties` that does not always return exactly `v` after `withLinktimeProperties(v)` has been called.

There are several possible solutions to this issue:
* Rename `withLinktimeProperties` to `withCustomLinktimeProperties`, and add a getter `customLinktimeProperties` that returns exactly the custom ones, or
* Change `linktimeProperties` so that it returns exactly what was passed in `withLinktimeProperties`, and add the predefined properties later in the pipeline (not in `NativeConfig`).

I think the second option is better, as it removes any logic from `NativeConfig`. It is a data bag and nothing else.

Reply:
Ok, that's sound fine. Currently, there is only one place where we use link-time properties (`trait LinktimeValueResolver`). 
I'm wondering what would be the best place to define predefined values in. If we would leave them in `object NativeConfig` we would be able to show resolved values in sbt using `show nativeConfig` which might be useful for debugging. On the other hand, since those would be injected anyway maybe we should not show them at all?

Reply:
I think it's fine not to show them at all at the sbt level.